web/concrete/src/Asset/JavascriptAsset.php

```diff
@@ -1,80 +1,95 @@
 <?php
 namespace Concrete\Core\Asset;
+
 use HtmlObject\Element;
 use Config;
 
-class JavascriptAsset extends Asset {
+class JavascriptAsset extends Asset
+{
+
+    protected $assetSupportsMinification = true;
+    protected $assetSupportsCombination = true;
 
-	protected $assetSupportsMinification = true;
-	protected $assetSupportsCombination = true;
+    protected $scriptType = 'text/javascript';
```

Comment:
You're missing the idea, you need to do nothing other than set the type attribute in your subclass's constructor.. Literally

```php
public function __construct()
{
    $this->setAttribute('type', 'text/template');
}
```

Here's an example of the commit I'd make if I wanted to add a TemplateAsset class:
```php
<?php
namespace Concrete\Core\Asset;

class TemplateAsset extends JavascriptInlineAsset
{

    protected $templateClass;
    
    public function __construct($class, $handle)
    {
        parent::__construct($handle);
        $this->templateClass = $class;
    }
    
    public function getAssetDefaultPosition()
    {
        return Asset::ASSET_POSITION_HEADER;
    }

    public function getAssetType()
    {
        return 'javascript-inline';
    }

    public function __toString()
    {
        return '<script class="' . $this->templateClass . '" type="text/template">' . $this->getAssetURL() . '</script>';
    }

}
```

That's all it would require.